### PR TITLE
Add brain_trait threshold evaluation with diagnostics and API exposure

### PR DESCRIPTION
**Pull request by leynos**
## Summary
- Adds a complete brain_trait threshold evaluation path with diagnostics, tests, and docs. This mirrors the brain_type pattern and exposes a pure, testable evaluation flow without rustc-private dependencies.

Key features:
- Threshold-driven evaluation for brain_trait with clearly defined rules (deny vs. warn)
- Diagnostic surface carrying measured values and formatted messages
- Builder-based configuration for thresholds (BrainTraitThresholdsBuilder)
- Comprehensive unit tests and BDD tests for end-to-end behavior
- Documentation updates and roadmap alignment

## Changes
- New modules for brain trait metrics
  - common/src/brain_trait_metrics/evaluation.rs
    - BrainTraitDisposition enum: Pass, Warn, Deny
    - BrainTraitThresholds struct with thresholds for warn/deny and CC warning
    - BrainTraitThresholdsBuilder with fluent setters and build()
    - evaluate_brain_trait(metrics, thresholds) -> BrainTraitDisposition
    - Internal helpers: total_method_count(), is_deny_triggered(), is_warn_triggered()
    - Re-exports for convenience to keep public API clean
  - common/src/brain_trait_metrics/diagnostic.rs
    - BrainTraitDiagnostic struct carrying trait name, disposition, method counts, CC sum, total items, implementor burden
    - BrainTraitDiagnostic::new(metrics, disposition)
    - format_primary_message(diagnostic), format_note(diagnostic), format_help(diagnostic)
- Evaluation and diagnostics tests
  - common/src/brain_trait_metrics/evaluation_tests.rs
  - common/src/brain_trait_metrics/diagnostic_tests.rs
- Behaviour-driven tests and feature
  - common/tests/features/brain_trait_evaluation.feature
  - common/tests/brain_trait_evaluation_behaviour.rs
- Public API surface updated
  - common/src/brain_trait_metrics/mod.rs: declare diagnostic and evaluation modules and re-exports
  - common/src/lib.rs: re-export BrainTraitDiagnostic, BrainTraitDisposition, BrainTraitThresholds, BrainTraitThresholdsBuilder, evaluate_brain_trait (format_* functions remain in evaluation module to avoid collisions with brain_type)
- Documentation and planning updates
  - docs/brain-trust-lints-design.md: implementation decisions for 6.3.2 added
  - docs/execplans/6-3-2-brain-trait-evaluation.md: detailed execution plan added
  - docs/roadmap.md: 6.3.2 checkbox updated to done

## Rationale
- Aligns brain_trait evaluation with brain_type (6.2.2) for consistency and reusability
- Keeps evaluation logic pure and compiler-independent, suitable for lint driver usage
- Deny is OR-based (method-count threshold) and takes precedence over Warn (AND-based on method-count and CC)
- Methods threshold uses only actual methods (required + default); associated items are intentionally excluded from the threshold calculations
- Diagnostic formatting surfaces measured values clearly for lint output

## Tests
- Unit tests cover:
  - Default threshold values and builder overrides
  - Pass/Warn/Deny boundaries including edge cases like zero defaults and CC-only signals
  - Associated items do not count toward method thresholds
- BDD tests (rstest-bdd) validate end-to-end scenarios against the feature file, including:
  - Trait within limits (Pass)
  - All warn conditions (Warn)
  - Many methods with low CC (Pass)
  - High CC alone (Pass)
  - Deny triggered by method count (Deny)
  - Deny supersedes warn when both signals present
  - Associated items excluded from method count (Pass)
  - Diagnostic surfaces measured values (CC, method counts, trait name)

## Validation plan
- [x] cargo fmt and lint pass
- [x] cargo test passes (existing tests preserved; new tests added)
- [x] Integration points wired without rustc_private dependencies
- [x] Public API surface clarified and re-exports updated

## Notes for reviewers
- Formatting functions (format_primary_message, format_note, format_help) are exposed via brain_trait_metrics::evaluation:: to avoid collision with brain_type re-exports at the crate root.
- The diagnostic and evaluation tests live in parallel files to keep source size manageable and maintainable.
- The feature file and behaviour tests provide end-to-end coverage for the threshold logic and diagnostic rendering.

If you want to tweak thresholds, you can use BrainTraitThresholdsBuilder::new().methods_warn(x).methods_deny(y).default_cc_warn(z).build() to tailor the rules per project needs.

◳ Generated by [DevBoxer](https://www.devboxer.com) ◰

---
ℹ️ Tag @devboxerhub to ask questions and address PR feedback

📎 **Task**: https://www.devboxer.com/task/287e5de9-4919-469e-ad75-013239190e84

### CODE REVIEW

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix:

[common/src/brain_trait_metrics/evaluation_tests.rs](https://github.com/leynos/whitaker/pull/163/files/688740c6d5633c92a1eb73de23aa2fc3b10ae8bd#diff-5cd2f75caebf3608882515dff7a867b6d8d4a25da4f50ddcd59d22cffbf9b7b9)

Comment on file

~~~rust
fn custom_methods_warn_threshold() {
    // 15 methods total (10+5), CC sum = 5*8 = 40.
    let metrics = build_trait_metrics("Custom", 10, 5, 8);
    let thresholds = BrainTraitThresholdsBuilder::new().methods_warn(15).build();
    assert_eq!(
        evaluate_brain_trait(&metrics, &thresholds),
        BrainTraitDisposition::Warn
    );
}
~~~

❌ New issue: [Code Duplication](https://codescene.io/projects/71836/delta?repo-id=1205552&review-id=163&biomarker=Code+Duplication&filename=common%2Fsrc%2Fbrain_trait_metrics%2Fevaluation_tests.rs&method=)
The module contains 3 functions with similar structure: custom_cc_warn_threshold,custom_deny_threshold,custom_methods_warn_threshold

File: common/tests/brain_trait_evaluation_behaviour.rs [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+    let result = evaluate_brain_trait(&metrics, &thresholds);
+    world.disposition.set(Some(result));
+    *world.built_metrics.borrow_mut() = Some(metrics);
+}
+
+#[when("the diagnostic message is formatted")]
+#[expect(
+    clippy::expect_used,
+    reason = "metrics and disposition are required for this behaviour test"
+)]
```
[the remark this answers is not in the document]
Comment (leynos) [reply common/tests/brain_trait_evaluation_behaviour.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

[the remark this answers is not in the document]
Comment (leynos) [reply common/tests/brain_trait_evaluation_behaviour.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.
